ENG-36631 Resource Manager sequential IP allocation strategy(4.2.12) (#63)
NSO-developer/nso-gitbook: platform-tools/resource-manager/resource-manager-api-guide.md

<figure><img src="../../.gitbook/assets/image.png" alt="" width="560"><figcaption></figcaption></figure>

> The synchronized allocation API request uses a Reactive-Fast-Map to allocate resources and still manages the interface to look synchronous. This means that as you create any allocation request from Northbound, you can see the allocation results, such as the requested IP subnet/ID in the same transaction. If a NB is making an allocation request, and in the same transaction a configuration is being applied to a specific device, the commit dry run receives the request response, and the response is processed by the Resource Manager and the configurations are pushed to the device in the same transaction. Thus, the NB user can see the get modification in the commit dry run.

Author: Addressed

The APIs exposed by RM are implemented in Python as well as Java, so the NB user can configure the service to be a Java package or a Python package and call the allocator API as per the implementation. The NB user can also use NSO CLI to make an allocation request to the IP allocator RM package.

> The IP resource pool supports two types of allocation method, named 'firstfree' and 'sequential'. We can specify the allocation method by setting the parameter `allocation-method` for an IP pool. The default allocation-method is 'firstfree' which is also a legacy allocation method, where released IP subnet can be reused immediately. While in the 'sequential' allocation method, released subnets are stored separately in a `available-secondary` set. when an allocation request is made and the requested subnet is not present in the available set, then the subnets can be allocated from the `available-secondary` set.

Reviewer: ```suggestion
The IP resource pool supports two types of allocation methods, namely `firstfree` and `sequential`. You can specify the allocation method by setting the parameter `allocation-method` for an IP pool. The default allocation method is `firstfree` (legacy allocation method), where a released IP subnet can be reused immediately, while in the `sequential` allocation method, the released subnets are stored separately in a `available-secondary` set. When an allocation request is made and the requested subnet is not present in the available set, the subnets can be allocated from the `available-secondary` set.
```

> By default the `available-secondary` set is hidden, and user must run the command `unhide debug` to view the details of the available-secondary set of an IP pool.

Author: Addressed
